lib/rules/no-whitespace-before-property.js

```diff
@@ -45,6 +45,16 @@ module.exports = {
         }
 
         /**
+        * Determines whether a MemberExpression needs an extra space after the object. For example, if a dot is placed after
+        an integer literal, it will be parsed as a decimal point rather than a property accessor.
+        * @param {ASTNode} node - the MemberExpression ode to check
+        * @returns {boolean} `true` if the node needs an extra space.
+        */
+        function needsExtraSpace(node) {
+            return !node.computed && node.object.type === "Literal" && typeof node.object.value === "number" && /^\d*$/.test(node.object.raw);
```

Comment:
Would it be worth adding this to `ast-utils.js` as an `isDecimalInteger()` function? There are a few other rules (e.g. https://github.com/eslint/eslint/pull/7186) that also need to run this check for numbers.